### ISSUE TITLE
allow setting the temperature unit to Fahreheit

### ISSUE THREAD

**Issue by aa755** [mac-zhou/midea-msmart#8]
I tried `self.data[0x09] |= 0x04` but it did not work. My AC still shows the temperature in Celcius. I can use the nethome app to set it to Fahrenheit. But because `device.refresh()` [doesn't work for me](https://github.com/mac-zhou/midea-msmart/issues/4#issuecomment-636548837), I cannot reverse engineer the state. How did you reverse engineer the LAN protocol?

**Comment by aa755** [comment]
`device.refresh()` works for me in the [cloud version](https://github.com/andersonshatch/midea-ac-lib/). I did the following experiment in an attempt to reverse engineer how the temperature unit is encoded:
I opened the nethome app in and set the temperature unit to Celcius and target temperature to 19 in heating mode. Then I logged the cloud response for `device.refresh()`:
````
5a5a01005700208000000000000000000000000063f50000000e0000000000000000000001000000aa1eac00000000000002c00183667f7f000000000069620000000000008d3400000000000000000000000000000000
````
The, I changed the temperature unit to Fahrenheit and did not change anything else. (the app automatically converted the 19C to 66F). Then I logged the cloud response for `device.refresh()`:
````
5a5a01005700208000000000000000000000000063f50000000e0000000000000000000001000000aa1eac00000000000002c00183667f7f00000000046962000000000000893400000000000000000000000000000000
````
Maybe this is useful in figuring out how to encode temperature units and fahrenheit temperatures.

**Comment by 0xbw** [comment]
> `device.refresh()` works for me in the [cloud version](https://github.com/andersonshatch/midea-ac-lib/). I did the following experiment in an attempt to reverse engineer how the temperature unit is encoded:
> I opened the nethome app in and set the temperature unit to Celcius and target temperature to 19 in heating mode. Then I logged the cloud response for `device.refresh()`:
> 
> ```
> 5a5a01005700208000000000000000000000000063f50000000e0000000000000000000001000000aa1eac00000000000002c00183667f7f000000000069620000000000008d3400000000000000000000000000000000
> ```
> 
> The, I changed the temperature unit to Fahrenheit and did not change anything else. (the app automatically converted the 19C to 66F). Then I logged the cloud response for `device.refresh()`:
> 
> ```
> 5a5a01005700208000000000000000000000000063f50000000e0000000000000000000001000000aa1eac00000000000002c00183667f7f00000000046962000000000000893400000000000000000000000000000000
> ```
> 
> Maybe this is useful in figuring out how to encode temperature units and fahrenheit temperatures.

Data is not encrypted from your log. first 40 bytes is the header.
the data is between 40 and 72, and the last is the checksum data.
I may know the problem about your device's "timeout."

**Comment by 0xbw** [comment]
> 我尝试过，`self.data[0x09] |= 0x04`但是没有用。我的交流电仍显示摄氏温度。我可以使用nethome应用将其设置为华氏度。但是因为`device.refresh()` [对我不起作用](https://github.com/mac-zhou/midea-msmart/issues/4#issuecomment-636548837)，所以我无法对状态进行逆向工程。您是如何对LAN协议进行反向工程的？

the decryption part is done.
the command part, you can see in the T0xAC.zip
https://community.home-assistant.io/t/any-support-for-midea-a-c/18742

**Comment by 0xbw** [comment]
i can try to change ```self.data[0x01] = 0x23```  to ```self.data[0x01] = 0x1e```in ```base_command```:

**Comment by aa755** [comment]
I figured out how to do this in the cloud version: https://github.com/andersonshatch/midea-ac-lib/pull/10/files#diff-6ad3d8b7b3d88f8f4c5bb4a84db572f8R131
Also, it may be worth using the +0.5C bit to allow finer temperature control, as I do in that PR.

**Comment by aa755** [comment]
solved by #9
